### PR TITLE
add asb metadata based on PR dapr/components-contrib#2203

### LINKED ISSUE #1942: Update docs with ASB metadata for bindings
**What content needs to be created or modified?**
Update docs with ASB metadata for bindings

**Describe the solution you'd like**

**Where should the new material be placed?**

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**

**Additional context**

### PR DESCRIPTION
**Pull request by mukundansundar**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Based on PR https://github.com/dapr/components-contrib/pull/2203

Details added based off of https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-azure-servicebus-queues/#message-metadata

## Issue reference

closes #1942

### CODE REVIEW

**Comment by mukundansundar** [comment]
@ItalyPaleAle Please review this PR.

**Review by ItalyPaleAle — APPROVED** [review]

File: daprdocs/content/en/reference/components-reference/supported-bindings/servicebusqueues.md [hunk 1]
```diff
@@ -129,6 +129,44 @@ This component supports **output binding** with the following operations:
 
 - `create`: publishes a message to the specified queue
 
+## Message metadata
+
+Azure Service Bus messages extend the Dapr message format with additional contextual metadata. Some metadata fields are set by Azure Service Bus itself (read-only) and others can be set by the client when publishing a message through `Invoke` binding call with `create` operation.
+
+### Sending a message with metadata
+
+To set Azure Service Bus metadata when sending a message, set the query parameters on the HTTP request or the gRPC metadata as documented [here](https://docs.dapr.io/reference/api/pubsub_api/#metadata).
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/servicebusqueues.md]:
```suggestion
To set Azure Service Bus metadata when sending a message, set the query parameters on the HTTP request or the gRPC metadata, as documented [here]({{< ref "pubsub_api.md#metadata" >}}).
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/servicebusqueues.md [hunk 2]
```diff
@@ -129,6 +129,44 @@ This component supports **output binding** with the following operations:
...
+- `metadata.SessionId`
+- `metadata.Label`
+- `metadata.ReplyTo`
+- `metadata.PartitionKey`
+- `metadata.To`
+- `metadata.ContentType`
+- `metadata.ScheduledEnqueueTimeUtc`
+- `metadata.ReplyToSessionId`
+
+> **Note:** The `metadata.MessageId` property does not set the `id` property of the cloud event returned by Dapr and should be treated in isolation.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/servicebusqueues.md]:
```suggestion
{{% alert title="Note" color="primary" %}}
- The `metadata.MessageId` property does not set the `id` property of the cloud event returned by Dapr and should be treated in isolation.
- The `metadata.ScheduledEnqueueTimeUtc` property supports the [RFC1123](https://www.rfc-editor.org/rfc/rfc1123) and [RFC3339](https://www.rfc-editor.org/rfc/rfc3339) timestamp formats.
{{% /alert %}}
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/servicebusqueues.md [hunk 3]
```diff
@@ -129,6 +129,44 @@ This component supports **output binding** with the following operations:
...
+
+> **Note:** The `metadata.ScheduledEnqueueTimeUtc` property supports the [RFC1123](https://www.rfc-editor.org/rfc/rfc1123) and [RFC3339](https://www.rfc-editor.org/rfc/rfc3339) timestamp formats.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/servicebusqueues.md]:
```suggestion
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/servicebusqueues.md [hunk 4]
```diff
@@ -129,6 +129,44 @@ This component supports **output binding** with the following operations:
...
+
+### Receiving a message with metadata
+
+When Dapr calls your application, it will attach Azure Service Bus message metadata to the request using either HTTP headers or gRPC metadata.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/servicebusqueues.md]:
```suggestion
When Dapr calls your application, it attaches Azure Service Bus message metadata to the request using either HTTP headers or gRPC metadata.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/servicebusqueues.md [hunk 5]
```diff
@@ -129,6 +129,44 @@ This component supports **output binding** with the following operations:
...
+
+- `metadata.DeliveryCount`
+- `metadata.LockedUntilUtc`
+- `metadata.LockToken`
+- `metadata.EnqueuedTimeUtc`
+- `metadata.SequenceNumber`
+
+To find out more details on the purpose of any of these metadata properties, please refer to [the official Azure Service Bus documentation](https://docs.microsoft.com/rest/api/servicebus/message-headers-and-properties#message-headers).
+
+> Note: that all times are populated by the server and are not adjusted for clock skews.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/servicebusqueues.md]:
```suggestion
{{% alert title="Note" color="primary" %}}
All times are populated by the server and are not adjusted for clock skews.
{{% /alert %}}
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
some formatting edits

File: daprdocs/content/en/reference/components-reference/supported-bindings/servicebusqueues.md [hunk 6]
```diff
@@ -129,6 +129,47 @@ This component supports **output binding** with the following operations:
 
 - `create`: publishes a message to the specified queue
 
+## Message metadata
+
+Azure Service Bus messages extend the Dapr message format with additional contextual metadata. Some metadata fields are set by Azure Service Bus itself (read-only) and others can be set by the client when publishing a message through `Invoke` binding call with `create` operation.
+
+### Sending a message with metadata
+
+To set Azure Service Bus metadata when sending a message, set the query parameters on the HTTP request or the gRPC metadata as documented [here]({{< ref "pubsub_api.md#metadata" >}}).
```
Comment (mukundansundar) [inline daprdocs/content/en/reference/components-reference/supported-bindings/servicebusqueues.md]: @ItalyPaleAle just a quick clarification, I believe we can set the metadata in the bindings request similar to how we set for pubsub right?

**Review by hhunter-ms — APPROVED** [review]

File: daprdocs/content/en/reference/components-reference/supported-bindings/servicebusqueues.md (hunk 6 above)
Comment (ItalyPaleAle) [reply daprdocs/content/en/reference/components-reference/supported-bindings/servicebusqueues.md]: Yes it should be similar: https://docs.dapr.io/reference/api/bindings_api/#invoking-output-bindings

Comment (ItalyPaleAle) [reply daprdocs/content/en/reference/components-reference/supported-bindings/servicebusqueues.md]: Although that link there is for pubsub and should eb changed to the bindings one

Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/servicebusqueues.md]:
```suggestion
To set Azure Service Bus metadata when sending a message, set the query parameters on the HTTP request or the gRPC metadata as documented [here]({{< ref "bindings_api.md" >}}).
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
just one comment per Alessandro's review

**Review by hhunter-ms — APPROVED** [review]

**Pull request merged**